relay-server/src/actors/processor.rs

```diff
@@ -2795,14 +2795,20 @@ impl Service for EnvelopeProcessorService {
                 return;
             };
 
+            let mut shutdown = false;
+
             loop {
                 let next_msg = async { tokio::join!(rx.recv(), semaphore.clone().acquire_owned()) };
 
                 tokio::select! {
                    biased;
 
-                    // TODO(iker): deal with the error when the sender of the channel is dropped.
-                    _ = subscription.changed() => self.global_config = subscription.borrow().clone(),
+                    update = subscription.changed(), if !shutdown => {
```

Comment:
From the [docs](https://docs.rs/tokio/latest/tokio/sync/watch/struct.Receiver.html#method.changed):

> This method returns an error if and only if the [`Sender`](https://docs.rs/tokio/latest/tokio/sync/watch/struct.Sender.html) is dropped.